executor/runtime/docker/docker.go

```diff
@@ -1408,9 +1401,30 @@ func (r *DockerRuntime) Start(parentCtx context.Context, pod *v1.Pod) (string, *
 		return "", nil, statusMessageChan, err
 	}
 
+	for _, c := range r.c.ExtraPlatformContainers() {
+		entry.Debugf("Adding status mmonitor for %s container (cid %s)", c.Name, c.ID)
+		eventChan, eventErrChan := r.getDockerEventsChannelsForContainer(eventCtx, c.ID)
+		go r.statusMonitor(eventCancel, c.ID, eventChan, eventErrChan, statusMessageChan)
+	}
+	for _, c := range r.c.ExtraUserContainers() {
+		entry.Debugf("Adding status mmonitor for %s container (cid %s)", c.Name, c.ID)
+		eventChan, eventErrChan := r.getDockerEventsChannelsForContainer(eventCtx, c.ID)
```

Comment:
Is there going to be many getDockerEventsChannelsForContainers? Should we instead have one channel, and multiplex it? Anyone know about the internals of how Docker events work? CC: @alekar

Reply:
Great question. 
The docs do say that having multiple filters on a stream (multiple container ids for us) is treated like an OR.
Having a single stream is probably cheaper for docker to do, I think that is a good idea.

The code that has to read the events has to decode "which" container emitted it anyway, might as well be a single event stream.

Reply:
Yea I'm going to go with the 1 channel + 1 docker event stream approach